### PR TITLE
Lint Code with JSCS

### PR DESCRIPTION
**Pull request by seanpdoyle**
* For now, copy `.jscsrc` from `emberjs/ember.js`
* Integrate into `npm start`

### CODE REVIEW

**Comment by seanpdoyle** [comment]
For now, the violations are only displayed in the output from `npm start`.

Violations will **NOT** break the build.

Any ideas on how to integrate `jscs` into `npm test`?

**Comment by rwjblue** [comment]
One way would be to change https://github.com/emberjs/data/blob/master/package.json#L21 to be `jscs packages && testem -R dot ci` or something.

File: package.json [hunk 1]
```diff
@@ -18,7 +18,7 @@
     "build:production": "ember build --environment=production",
     "prepublish": "bower install && npm run-script build:production",
     "start": "ember serve",
-    "test": "testem -R dot ci",
+    "test": "node_modules/broccoli-jscs/node_modules/jscs/bin/jscs packages && testem -R dot ci",
```
Comment (seanpdoyle) [inline package.json]: @rwjblue this seems a little extreme, but `jscs` isn't a direct dependency.

Is this better or worse than adding `jscs` to the dependencies and executing it directly?

Should I add `npm install -g jscs` to `.travis.yml`?

**Comment by seanpdoyle** [comment]
@igorT @fivetanley sorry to bug you, but since this PR touches almost every file in the codebase, it's somewhat blocking to future refactorings.

If I could get your :eyes: on it ASAP, I could keep cleaning up the test suite.

**Comment by seanpdoyle** [comment]
Not sure how to go about making the windows CI green

**Comment by stefanpenner** [comment]
@seanpdoyle it doesn't work quite yet (we could use the help if u have time)

this looks great!

**Pull request merged**